queue_processors: Link to SETTINGS/DATA EXPORTS from data export completion notification

Esther146 opened the pull request:
This commit changes the url of the notification message to point to settings/data exports instructing the user to follow the link to download the exported data.

Fixes:#26923


**Screenshots and screen captures:**
<img width="883" alt="Screenshot 2023-10-05 at 23 06 18" src="https://github.com/zulip/zulip/assets/96547471/456ff8cf-56d6-458c-b544-d74a4a090cbf">
<img width="886" alt="Screenshot 2023-10-10 at 22 58 04" src="https://github.com/zulip/zulip/assets/96547471/2c2cfda6-14c0-404e-b706-771801d2d6de">



<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
There is a string already specified in the issue description. We should use it.

Esther146 commented:
> There is a string already specified in the issue description. We should use it.


Hi @alya , It looks like the link in the original issue is not going to the data export settings. The one I have on the PR redirects to the data export settings. Can you check if I'm not missing anything. Thanks

Esther146 commented:
> There is a string already specified in the issue description. We should use it.

Hi @alya , I didn’t understand the issue description completely earlier but now I think I have made the required changes.

karlstolley commented:
@Esther146 by "string," Alya means the message that accompanies the download link. In the original issue, that's `Your data export is complete. [View and download exports.](/#organization/data-exports-admin)` (though probably not with the Markdown link syntax). Don't invent your own, in other words.

Be sure to rewrite your commit history to match our style and expectations as outlined in https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html

karlstolley commented:
Looks pretty good, @Esther146! Please update your screenshot to reflect the changes you've made.

Esther146 commented:
> Looks pretty good, @Esther146! Please update your screenshot to reflect the changes you've made.

Hi @karlstolley, Please do look at it again. I have added the updated screenshot. 
Thank you.

alya commented:
Looks good to me in manual testing!

timabbott commented on zerver/worker/queue_processors.py:
```diff
@@ -1112,9 +1112,12 @@ def failure_processor(event: Dict[str, Any]) -> None:
-                    "Your data export is complete and has been uploaded here:\n\n{public_url}"
-                ).format(public_url=public_url)
+                export_settings_link = (
+                    "[View and download exports](#organization/data-exports-admin)"
```
This breaks internationalization. Read https://zulip.readthedocs.io/en/latest/translating/internationalization.html to see why this approach doesn't work.

timabbott commented:
I rewrote the commit to fix the i18n issues, and also rewrote the commit message to explain the motivation for the change. @Esther146 thanks for the contribution, and please take the time to study the changes I made -- we'll want to see you've learned in your next PRs.

Esther146 commented:
> I rewrote the commit to fix the i18n issues, and also rewrote the commit message to explain the motivation for the change. @Esther146 thanks for the contribution, and please take the time to study the changes I made -- we'll want to see you've learned in your next PRs.

Thank you @timabbott, I will definitely study it and take my corrections.
